### DISCUSSION TITLE
Swift 5.9 and C++ interoperability

### DISCUSSION THREAD

**Discussion by VojtaStruhar** [migueldeicaza/SwiftGodot#162] [General]
Today I read the release notes for Swift 5.9 and they mentioned [bidirectional C++ interoperability](https://www.swift.org/blog/swift-5.9-released/#ecosystem). I immediately thought of this project. In the README there is a mention of needing to write some [glue code](https://github.com/migueldeicaza/SwiftGodot#your-swift-code) and I was wondering:

- Is the (new) interoperability going to impact this project in any way? If so, how?

To me it seems REALLY exciting! I'd LOVE to write games with Swift and this sounds to me like it would make the job of this project a whole lot easier.

Thoughts?

**Comment by migueldeicaza** [comment]
Not in the short term, the way that Godot works both for its internal workflow and to communicate with external languages is via a system called GDExtension.  This is where the ABI/API for Godot is defined 

SwiftGodot is another client of this API, and while you could skip GDExtension and go straight to C++ you would do that against an unstable api and without plugging where GDExtension is.

**Reply by VojtaStruhar** [reply]
Thanks for the reply. I absolutely agree that skipping GDExtension is a bad idea. 

Does this change make interacting with GDExtension at least more convenient? Maybe skipping some boilerplate or _glue code_?

**Reply by migueldeicaza** [reply]
It doesn’t help in this particular case, because the Godot folks went out of their way to design a system that could work for many languages.  This is rare in our industry.

that said, it would be useful for those of us that have a urge to rewrite everything in Swift, and I would hope that one day Godot moves parts of it to it :-)

**Reply by VojtaStruhar** [reply]
Good to know! Thanks!
